spec/inputs/jdbc_spec.rb

```diff
@@ -113,4 +114,59 @@
 
     @database.drop_table(:test_table)
   end
+
+  context "persistence" do
+
+    it "should respect last run metadata" do
+      settings = {
+        "statement" => "SELECT * FROM SYSIBM.SYSDUMMY1",
+        "last_run_metadata_path" => Stud::Temporary.pathname
+      }
+      plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
+      plugin.register
+      q = Queue.new
+      plugin.run(q)
+      plugin.teardown
+
+      insist { File.exists?(settings["last_run_metadata_path"]) }
+      last_run = YAML.load(File.read(settings["last_run_metadata_path"]))
+
+      plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
+      plugin.register
+      insist { plugin.instance_variable_get("@sql_last_start") } == last_run
+      plugin.teardown
+    end
+
+    it "should ignore last run metadata if :clean_run set to true" do
+      settings = {
+        "statement" => "SELECT * FROM SYSIBM.SYSDUMMY1",
+        "last_run_metadata_path" => Stud::Temporary.pathname,
+        "clean_run" => true
+      }
+
+      File.open(settings["last_run_metadata_path"], "w") { |f| f.write(YAML.dump(Time.at(1).utc)) }
+
+      plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
+      plugin.register
+
+      insist { plugin.instance_variable_get("@sql_last_start") } == Time.at(0).utc
+
+      plugin.teardown
+    end
+
+    it "should not save state if :record_last_run is false" do
+      settings = {
+        "statement" => "SELECT * FROM SYSIBM.SYSDUMMY1",
+        "last_run_metadata_path" => Stud::Temporary.pathname,
+        "record_last_run" => false
+      }
+
+      plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
+      plugin.register
+
+      insist { File.exists?(settings["last_run_metadata_path"]) }
```

Comment:
This doesn't check anything (insist's block return value is not used). Insist has a `reject` to do the opposite. Also this should be acting on the `File` -

```ruby
reject { File }.exists?(settings["last_run_metadata_path"])
```

Then again, we want to move to using rspec 3's expect(), so this is preferred:

```ruby
expect(File).to(be_exists?(settings["last_run_metadata_path"]))
```

(I think this is preferred, anyway? There may be a different way to express it that doesn't read as awkwardly as "be exists"